### DISCUSSION TITLE
Namespace visibility for types

### DISCUSSION THREAD

**Discussion by kovianyo** [dotnet/csharplang#4559] [Language Ideas] [answered]
Hi all,

I am missing in C# a way to limit the visiblitiy of types to their namespace and child namespaces. 

What I want to achieve is the follwing: to have a folder (corresponding to a namespace) wher one class is **internal**, so it is visible from elswhere in the project, but other classes with a proposed "namespace visibility" in this same folder (or subfolders, thus in child namespaecs) are only visible inside this folder (and subfolders), but not outside this folder (also not in the same project). So these classes would be visible only in this folder, and the **internal** class would be an entry point.

This would enable an encapsulation what is between "nested private tpyes" and "creating a separate assembly".

What do you think of this? Or is there a way to achieve this what I am missing?

thank you,
Tamas

**Comment by dstarkowski** [comment, marked as answer]
Duplicate: #4153, #1138 and possibly others
